src/planner/macos/profile_queries.rs

```diff
@@ -0,0 +1,123 @@
+use crate::planner::macos::profiles::{
+    HardDiskInternalOpts, MountControls, Policies, Profile, ProfileItem, SystemUIServer, Target,
+};
+
+struct TargetProfileItem<'a> {
+    target: &'a Target,
+    profile: &'a Profile,
+    item: &'a ProfileItem,
+}
+
+pub struct TargetProfileHardDiskInternalOpts<'a> {
+    pub target: &'a Target,
+    pub profile: &'a Profile,
+    pub opts: &'a Vec<HardDiskInternalOpts>,
```

Comment:
If you're gonna give these lifetimes, then let's go all the way and use a slice instead of a `Vec`

```suggestion
    pub opts: &'a [HardDiskInternalOpts],
```